scripts/encodings_retrival/extract_scGPT_encodings.py

```diff
@@ -131,23 +161,42 @@ def load_scgpt_embedding(model_dir):
 
 @click.command()
 @click.option(
-    "--model_dir",
-    type=click.STRING,
-    help="directory with scGPT model and gene vocabulary",
-    default="/dccstor/bmfm-targets/ScGPT_weights/blood_model",
-    # model downloaded from https://drive.google.com/drive/folders/1kkug5C7NjvXIwQGGaGoqXTk_Lb_pDrBU
+    "--allow-downloads",
+    "-l",
+    type=click.BOOL,
+    help="download files directly from urls",
+    default=False,
 )
 @click.option(
-    "--output-file-name",
+    "--input-file-dir",
     type=click.STRING,
-    help="The output csv file name",
-    default="/dccstor/bmfm-targets/ScGPT_weights/blood_model_embedding.csv",
+    help="The path to the directory with the data files, if no files exists and the user will download from a url, this is the path to the directory the files will be saved to",
+    default="./encodings",
+)
+@click.option(
+    "--model-type",
+    "-m",
+    type=click.Choice(["human", "blood"]),
+    help="type of ScGPT model to use, can be either 'human' or 'blood'.",
+    default="human",
 )
-def main(model_dir, output_file_name):
-    print(f"Reading scGPT model from {model_dir}")
-    embedding = load_scgpt_embedding(model_dir=model_dir)
-    embedding.to_csv(output_file_name)
-    print(f"Saved to {output_file_name}")
+def main(allow_downloads, input_file_dir, model_type):
+
+    if model_type == "human":
+        model_urls = HUMAN_MODEL_URLS
+    elif model_type == "blood":
+        model_urls = BLOOD_MODEL_URLS
+
+    model_encodings_dir = Path(input_file_dir) / f"ScGPT-{model_type}"
```

Comment:
So you can put this line, the following, and the to CSV together in a dump encoding method.